### DISCUSSION TITLE
Decoupled setup - Razor Page routing works locally but not on server

### DISCUSSION THREAD

**Discussion by Jack-s** [OrchardCMS/OrchardCore#16094] [Q&A]
Using a decoupled setup everything works locally but when published (IIS) if OrchardCore is enabled I get a 404 for all my razor pages however the admin controller does work. If I comment out the `services.AddOrchardCms();` and instead use the `services.AddRazorPages();` then my pages work but the admin controller doesn't.

I'm not certain but I think the issue only started after I imported a content type that has a couple of media field content parts (this is the only content type I created and I haven't enabled any other features). 

For some more context I'm just referencing the `OrchardCore.Application.Cms.Core.Targets` package and my startup.cs looks like the following:

        public void ConfigureServices(IServiceCollection services)
        {
            services.AddOrchardCms();
	    }        
		
		
	     public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }
            else
            {
                app.UseExceptionHandler("/Error");
                app.UseHsts();
            }

            app.UseHttpsRedirection();
            app.UseStaticFiles();
            app.UseOrchardCore();
        }

**Comment by hishamco** [comment]
Why do you need to call `services.AddRazorPages();` it has already been called on your behalf.

While I'm working heavily with decoupled CMS, I never faced such an issue locally or LIVE, please more details to reproduce what you're facing

**Comment by ns8482e** [comment]
Provide `@page` with the route url or write Conversions. 

If  `@page` doesn't have route url default path will be `/Module/Page`. If the page is in main web app then instead of Module name provide your main we project's name
